pkg/cli/cli.go

```diff
@@ -192,38 +190,34 @@ func (c *cli) initialize() error {
 		return err
 	}
 
-	// With or without a config, a user can:
+	// When invoking 'init', a user can:
 	// 1. Not set --plugins
-	// 2. Set --plugins to a plugin, ex. --plugins=go
-	// In case 1, either default plugins or the config layout can be used to
-	// determine which plugin to use. Otherwise, the value passed to --plugins
-	// is used. Since both layout and a --plugins value are user input, they
-	// may need to be resolved to known plugin keys.
-	var filterKeys map[string]string
+	// 2. Set --plugins to a plugin, ex. --plugins=go-x
+	// In case 1, default plugins will be used to determine which plugin to use.
+	// In case 2, the value passed to --plugins is used.
+	// For all other commands, a config's 'layout' key is used. Since both
+	// layout and --plugins values can be short (ex. "go/v2.0.0") or unversioned
+	// (ex. "go.kubebuilder.io") keys or both, their values may need to be
+	// resolved to known plugins by key.
+	plugins := c.pluginsFromOptions[c.projectVersion]
 	switch {
-	case len(c.cliPluginKeys) != 0:
-		// Filter plugins by keys passed in CLI.
-		filterKeys = c.cliPluginKeys
+	case c.cliPluginKey != "":
```

Comment:
I'd prefer len instead of != ""

Reply:
Both are idiomatic. It should be clear that `c.cliPluginKey` is a string, hence `!= ""`. Ditto below.

Reply:
the problem with != "" is that will not work for !=" " when should for example.